### PR TITLE
Encrypting and clearing workflow options.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: database/migration/src/main/scala/cromwell/database/migration/custom/BatchedTaskChange.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
+package cromwell.database.migration.custom
```
Comment (kshakir) [inline database/migration/src/main/scala/cromwell/database/migration/custom/BatchedTaskChange.scala]: ToL: I'm open to other names for this package. I stopped myself from throwing it in yet-another generic `*.util` package.

File: engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala [hunk 2]
```diff
@@ -117,6 +118,38 @@ case class WorkflowStoreActor(store: WorkflowStore, serviceRegistryActor: ActorR
...
+
+  private def processSources(sources: NonEmptyList[WorkflowSourceFiles],
+                             processOptions: WorkflowOptions => WorkflowOptionsJson):
+  Try[NonEmptyList[WorkflowSourceFiles]] = {
+    val nelTries: NonEmptyList[Try[WorkflowSourceFiles]] = sources map processSource(processOptions)
+    val seqTries: Seq[Try[WorkflowSourceFiles]] = nelTries.list.toList
+    val trySeqs: Try[Seq[WorkflowSourceFiles]] = TryUtil.sequence(seqTries)
+    import scalaz.Scalaz._
+    val tryNel: Try[NonEmptyList[WorkflowSourceFiles]] = trySeqs.map(_.toList.toNel.get)
+    tryNel
```
Comment (kshakir) [inline engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala]: ToL: Also open to folks helping with this HoF soup. Of course will be a different brand of soup for 0.22_hotfix and develop due to scalaz->cats.

**Comment by mcovarr** [comment]
:+1:

File: database/migration/src/main/scala/cromwell/database/migration/custom/BatchedTaskChange.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    */
+  def migrateBatchQuery: String
+
+  /**
+    * Migrate a row.
+    *
+    * Read the values from readRow, update the values, set the updated values on the migrateStatement, and then call
+    * migrateStatement.addBatch(). Return the (estimated) number of rows to be written by this batch.
+    *
+    * @param readRow          The row to migrate
```
Comment (kshakir) [inline database/migration/src/main/scala/cromwell/database/migration/custom/BatchedTaskChange.scala]: I thought about changing the name to better match the type (this code is copied from elsewhere). But I like that the name conveys that the implementor _is going to have a bad time_ if they call any other methods on the `ResultSet` besides accessing the current row values.

**Comment by geoffjentry** [comment]
Just to make sure I'm understanding correctly, all the stuff in the migration package is a one time run, correct?

👍

**Comment by kshakir** [comment]
Yes, migration runs once as a changeset, and liquibase records that changeset as done within its tables.

**Pull request merged**